### ISSUE TITLE
Refactor and standardize fetching and caching

### ISSUE THREAD

**Issue by Sturlen** [open5e/open5e#500]
Use an established library like [Tanstack Query](https://tanstack.com/query/latest/docs/framework/vue/overview) to replace the custom Pinia cache and data fetching solution. Standardizing the way data fetching is done throughout the site avoids a lot of bugs and edges cases and would make a move to the V2 api a lot smoother. I've used this solution before and am happy to implement it here if this is the direction you want to go in.

Example of what Tanstack Query syntax looks like:
```vue
<script setup>
import { useQueryClient, useQuery, useMutation } from '@tanstack/vue-query'

// Access QueryClient instance
const queryClient = useQueryClient()

// Query
const { isPending, isError, data, error } = useQuery({
  queryKey: ['todos'],
  queryFn: getTodos,
})

// Mutation
const mutation = useMutation({
  mutationFn: postTodo,
  onSuccess: () => {
    // Invalidate and refetch
    queryClient.invalidateQueries({ queryKey: ['todos'] })
  },
})

function onButtonClick() {
  mutation.mutate({
    id: Date.now(),
    title: 'Do Laundry',
  })
}
</script>

<template>
  <span v-if="isPending">Loading...</span>
  <span v-else-if="isError">Error: {{ error.message }}</span>
  <!-- We can assume by this point that `isSuccess === true` -->
  <ul v-else>
    <li v-for="todo in data" :key="todo.id">{{ todo.title }}</li>
  </ul>
  <button @click="onButtonClick">Add Todo</button>
</template>

```

**Comment by calumbell** [comment]
This is a great idea, defintely something worth sorting out before we switch over to API v2.

I have never come across Tanstack Query before. What are its main advantages over [Nuxt's native data fetching](https://nuxt.com/docs/getting-started/data-fetching)?

**Comment by Sturlen** [comment]
I've not used Nuxt's data fetching, but from what I can see it's more focused on server-side fetching. Tanstack Query works great with queries that depend on the results of other queries, such as fetching the list of sources before fetching the main content. It also supports any async methods such as axios, rather than only fetch.

**Comment by calumbell** [comment]
That is good to know, thanks for taking the time to explain. Nuxt native data fetching does work both client and server side, but from some very precursory digging around online it looks like Tanstack Query is much better at handling the caching/de-dupe for queries that depend on other queries. I'd say go for it!

Is this an issue you'd be up for working on? It looks like it will be quite a lot of work, as the Pinia store touches so many parts of the code-base. Happy to pitch in if I can help, or lead if you don't have bandwidth to work on this feature.

**Comment by Sturlen** [comment]
I've been doing some exploratory work and I should be able to handle it by myself. The plan would be to completely replace Pinia  with Tanstack Query since the two don't really mix.

**Comment by calumbell** [comment]
Awesome, good luck! I'll keep an eye open for the PR

**Comment by calumbell** [comment]
Resolved in PR #503
